src/pages/Admin/MapPreview/MapPreview.tsx

```diff
@@ -123,7 +124,7 @@ const MapPreview: React.FC<MapPreviewProps> = ({
           <Legend
             text="Edit background"
             position="right"
-            onClick={() => handleBackgroundRemove()}
+            onClick={handleBackgroundRemove}
```

Comment:
This should probably be done in an async thunk or updating some local state on error or similar; rather than just calling it and 'forgetting' about it. As is, we have no way of knowing if it worked/failed/etc, and reacting appropriately.

Reply:
I can just wrap the promise inside with try/catch and use some state. How does that sound?

Reply:
Yeah, that would be fine for now, that's a similar pattern to what we have elsewhere.

Reply:
I'm adding this as we agreed on it, but it feels weird and not correct. I think we should have a more global and generic way of handling these. maybe notifications or some kind of pop ups.

Reply:
https://github.com/sparkletown/internal-sparkle-issues/issues/630